src/main/java/Domain/ShopController.java

```diff
@@ -0,0 +1,60 @@
+package Domain;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ShopController {
+    private static ShopController _shopController;
+    private List<Shop> _shopsList;
+
+    private ShopController() {
+        this._shopsList = new ArrayList<>();
+    }
+
+    // Public method to provide access to the _shopController
+    public static synchronized ShopController getShopController() {
+        if (_shopController == null) {
+            _shopController = new ShopController();
+        }
+        return _shopController;
+    }
+
+    public Shop getShopByShopId(Integer shopId){
+        for (Shop shop : this._shopsList) {
+            if (shop.getShopId().equals(shopId)) {
+                return shop;
+            }
+        }
+        return null;
+    }
+
+    public Boolean isShopIdExist(Integer shopId)
+    {
+        for (Shop shop : this._shopsList) {
+            if (shop.getShopId().equals(shopId)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public void OpenNewShop(Integer shopId, String userName) throws Exception 
+    {
+        //TODO: need to implement  (check if shop exist)
```

Comment:
remove todo if done implemented

Reply:
add todo about permisions

Reply:
I think that each user the logged in as permission to open new shop

Reply:
thats true thanks! missed that